packages/cyberstorm/src/components/Forms/CreateTeamForm/CreateTeamForm.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+import { faArrowsRotate } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useController, useForm, Form } from "react-hook-form";
+import * as Button from "../../Button";
+import { TextInput } from "../../TextInput/TextInput";
+import { useToast } from "../../Toast/Provider";
+import styles from "./CreateTeamForm.module.css";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { isErrorResponse } from "../../../utils/type_guards";
+import { useEffect } from "react";
+
+export function CreateTeamForm() {
+  const toast = useToast();
+
+  // "Required field" error is different from value being too little
+  // Check that the value is "> 0"
+  const schema = z.object({
+    teamName: z
+      .string({ required_error: "Team name is required" })
+      .min(1, { message: "Team name is required" }),
+  });
+
+  const {
+    control,
+    formState: { isSubmitting, errors },
+  } = useForm<z.infer<typeof schema>>({
+    mode: "onSubmit",
+    resolver: zodResolver(schema),
+  });
+
+  const teamName = useController({
+    control: control,
+    name: "teamName",
+  });
+
+  // We have to do this because the RHF Form component is in beta and it
+  // doesn't have a callback prop like "onValidation" that could take in
+  // the generalized addToast error informing block.
+  useEffect(() => {
```

Comment:
IMO this is bad UX. The form error should be shown on the form, next to the field. Not as a toast.

Also if the beta status actually prevents us from using such basic features (I assume it actually doesn't), we should reconsider if it's something we want to use in this soon-to-be-production project.